core/controller/src/main/scala/whisk/core/controller/Meta.scala

```diff
@@ -105,15 +143,23 @@ trait WhiskMetaApi extends Directives with PostActionActivation {
             }
         }
 
-        def activate(action: WhiskAction) = {
+        def activate(action: WhiskAction): Future[(ActivationId, Option[WhiskActivation])] = {
             // precedence order for parameters:
             // package.params -> action.params -> query.params -> request.entity (body) -> augment arguments (namespace, path)
-            systemKey flatMap {
-                val content = requestParams ++ Map(
-                    "__ow_meta_verb" -> method.value.toLowerCase.toJson,
-                    "__ow_meta_path" -> restofPath.toJson,
-                    "__ow_meta_namespace" -> user.namespace.toJson)
-                invokeAction(_, action, Some(JsObject(content)), blocking = true, waitOverride = true)
+            systemKey flatMap { identity =>
+                val invokeParams = if (action.hasFinalParamsAnnotation) {
+                    requestParams -- action.parameters.immutableParameters
+                } else requestParams // in the absence of immutable annotations, return the request untouched
+
+                if (invokeParams.size == requestParams.size) {
```

Comment:
Yes but intersect could be more expensive so I opted for a size check instead against the original map.